src/modules/endpoint/sagas/__tests__/watchFetchCategory.spec.js

```diff
@@ -0,0 +1,249 @@
+        type: 'FETCH_CATEGORY',
+        params: {
+          city,
+          language,
+          path: initialPath,
+          depth: 2,
+          key: 'categories-key',
+          criterion: {
+            forceUpdate: false,
+            shouldRefreshResources: true
+          }
+        }
+      }
+
+      return expectSaga(fetchCategory, dataContainer, action)
+        .withState({
+          cityContent: {
+            city: city
+          }
+        })
+        .put({
+          type: 'PUSH_CATEGORY',
+          params: {
+            categoriesMap: categories,
+            resourceCache: resources,
+            path: initialPath,
+            cityLanguages: languages,
+            depth: 2,
+            key: 'categories-key',
+            language,
+            city
+          }
+        })
+        .run()
+    })
+
+    it('should put an action which pushes the categories when peeking', async () => {
+      const { categories, resources, dataContainer, initialPath } = await createDataContainer(city, language)
+      const anotherCity = 'anotherCity'
+
+      const action: FetchCategoryActionType = {
+        type: 'FETCH_CATEGORY',
+        params: {
+          city,
+          language,
+          path: initialPath,
+          depth: 2,
+          key: 'categories-key',
+          criterion: {
+            forceUpdate: false,
+            shouldRefreshResources: true
+          }
+        }
+      }
+
+      return expectSaga(fetchCategory, dataContainer, action)
+        .withState({
+          cityContent: {
+            city: anotherCity
+          }
+        })
+        .put({
+          type: 'PUSH_CATEGORY',
+          params: {
+            categoriesMap: categories,
+            resourceCache: resources,
+            path: initialPath,
+            cityLanguages: [],
+            depth: 2,
+            key: 'categories-key',
+            language,
+            city
+          }
+        })
+        .run()
+    })
+
+    it('should do nothing if content can not be retrieved (e.g. language invalid) and not peeking', async () => {
+      const { dataContainer, initialPath } = await createDataContainer(city, language)
+
+      const invalidLanguage = '??'
+      const action: FetchCategoryActionType = {
+        type: 'FETCH_CATEGORY',
+        params: {
+          city,
+          language: invalidLanguage,
+          path: initialPath,
+          depth: 2,
+          key: 'categories-key',
+          criterion: {
+            forceUpdate: false,
+            shouldRefreshResources: true
+          }
+        }
+      }
+
+      return expectSaga(fetchCategory, dataContainer, action)
+        .withState({
+          cityContent: {
+            city: city
+          }
+        })
+        .not.put.like({ action: { type: 'PUSH_CATEGORY' } })
+        .not.put.like({ action: { type: 'FETCH_CATEGORY_FAILED' } })
+        .run()
+    })
+
+    it('should try to loadCityContent with an invalid language when peeking', async () => {
```

Comment:
Right now, this is kind of the same test as the one below (should put an error action) with a different error message. Wouldn't it be enough to just assert `loadCityContent` was called (http://redux-saga-test-plan.jeremyfairbank.com/integration-testing/partial-matching.html) if you just want to test that it also gets called if the language is invalid while peeking?